java-source/src/main/java/com/example/api/ExampleApi.java

```diff
@@ -92,21 +94,33 @@ public Response createIOU(IOU iou, @PathParam("party") String partyName) throws
                 otherParty,
                 new IOUContract());
 
-        // The line below blocks and waits for the flow to return.
-        final ExampleFlow.ExampleFlowResult result = getOrThrow(services
-                .startFlowDynamic(ExampleFlow.Initiator.class, state, otherParty)
-                .getReturnValue(), null);
+        Response.Status status;
+        String msg;
+        try {
+            // The line below blocks and waits for the flow to return.
+            final SignedTransaction result = services
+                    .startFlowDynamic(ExampleFlow.Initiator.class, state, otherParty)
+                    .getReturnValue()
+                    .get();
 
-        final Response.Status status;
-        if (result instanceof ExampleFlow.ExampleFlowResult.Success) {
             status = Response.Status.CREATED;
-        } else {
+            msg = String.format("Transaction id %s committed to ledger.", result.getId());
+
+        } catch (Throwable ex) {
             status = Response.Status.BAD_REQUEST;
+
+            if (ex.getCause() instanceof FlowSessionException) {
+                msg = "Counterparty flow terminated unexpectedly.";
+            } else if (ex.getCause() instanceof RuntimeException) {
+                msg = ex.getCause().getMessage();
```

Comment:
I'm assuming here that you're catching `RuntimeException` because of the some valid verification failure. The problem is that it could also be some internal system error which we certainly don't want to be passing onto the REST client. Perhaps for we now we should introduce a custom exception which we throw in try-catch block around the relevant code.